OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift

```diff
@@ -10,118 +10,133 @@ import Foundation
+        case internal2BitVariableSetAndManipulatedInMainLoopRoutines2 = 0b0010
+        case internal2BitVariableSetAndManipulatedInMainLoopRoutines3 = 0b0100
+        case insulinStateTableCorruption                              = 0b1000
+        
+        public var description: String {
+            switch self {
+                case .none:
+                    return "None"
+                case .immediateBolusInProgress:
+                    return "Immediate Bolus In Progress"
+                case .internal2BitVariableSetAndManipulatedInMainLoopRoutines2:
+                    return "Internal 2-Bit Variable Set And Manipulated In Main Loop Routines 0x02"
+                case .internal2BitVariableSetAndManipulatedInMainLoopRoutines3:
+                    return "Internal 2-Bit Variable Set And Manipulated In Main Loop Routines 0x03"
+                case .insulinStateTableCorruption:
+                    return "Insulin State Table Corruption"
+            }
+        }        
+    }
+    
     public init(encodedData: Data) throws {
         
-        if encodedData.count < 19 {
+        if encodedData.count < 21 {
             throw MessageBlockError.notEnoughData
         }
         
-        guard let reservoirStatus = ReservoirStatus(rawValue: encodedData[1]) else {
-            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "ReservoirStatus")
+        guard PodProgressStatus(rawValue: encodedData[1]) != nil else {
+            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "PodProgressStatus")
         }
-        self.reservoirStatus = reservoirStatus
+        self.podProgressStatus = PodProgressStatus(rawValue: encodedData[1])!
         
         self.deliveryType = DeliveryType(rawValue: encodedData[2] & 0xf)
         
         self.insulinNotDelivered = podPulseSize * Double((Int(encodedData[3] & 0x3) << 8) | Int(encodedData[4]))
         
         self.podMessageCounter = encodedData[5]
-        self.unknownPageCode = Double(Int(encodedData[6]) | Int(encodedData[7]))
+        self.unknownPageCode = encodedData[6...7]
         
-        self.originalLoggedFaultEvent = FaultEventCode(rawValue: encodedData[8])
+        self.currentStatus = FaultEventCode(rawValue: encodedData[8])
         
-        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double((Int(encodedData[9] & 0b1) << 8) + Int(encodedData[10])))
+        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double(encodedData[9...10].toBigEndian(UInt16.self)))
         
-        self.insulinRemaining = podPulseSize * Double((Int(encodedData[11] & 0x3) << 8) | Int(encodedData[12]))
+        let resHighBits = Int(encodedData[11] & 0x03) << 6
+        let resLowBits = Int(encodedData[12] >> 2)
+        let reservoirValue = round(Double((resHighBits + resLowBits) * 50)/255)
+        if reservoirValue < StatusResponse.maximumReservoirReading {
```

Comment:
This should be set to nil, if we are not at the point where we have a valid reservoir reading.  This is already within the context of having data, so we do not need to distinguish between not being able to get status and this field being set to >50.

Reply:
Ok, changed this be set to nil.